framework/src/modules/chain/block_processor_v2.js

```diff
@@ -179,9 +206,13 @@ class BlockProcessorV2 extends BaseBlockProcessor {
 		// TODO: Remove validate new since it's no longer required
 		this.validateNew.pipe([() => Promise.resolve()]);
 
-		this.verify.pipe([data => this.blocksModule.verify(data)]);
+		this.verify.pipe([({ block }) => this.bft.verifyNewBlock(block)]);
```

Comment:
we don't need to verify block here anymore?
it feels bit weird to have `verify` in apply step

Reply:
The block must be broadcasted prior to performing any database related checks. That is what is described in the LIP. I concede that it looks odd to have verify in apply but perhaps we should restructure the blocks module a little bit to adjust to that?
Also, we opened this PR because as draft because we want to have a discussion on how to approach this topic - should we update the blocks module to reflect the LIP or should we go on a tangent from what is described in the LIP.
Please suggest alternatives. :)